src/core.js

```diff
@@ -478,7 +478,13 @@ jQuery.each("Boolean Number String Function Array Date RegExp Object Error".spli
 });
 
 function isArraylike( obj ) {
-	var length = obj.length,
+
+	// Support: iOS 8.2 (not reproducible in simulator)
+	// `in` check used to prevent JIT error (gh-2145)
+	// hasOwn isn't used here due to false negatives
+	// regarding Nodelist length in IE
+	// obj is casted to Object to avoid type errors (gh-2185)
+	var length = "length" in Object( obj ) && obj.length,
```

Comment:
Yea, good call. I assumed it worked with strings because I thought I had checked that a while back, but we don't have tests for it, so I'm not surprised it broke if it did work.